### PR TITLE
Proposal: Logical Assignment Operators

### PR DESCRIPTION
**Pull request by jridgewell**
https://github.com/jridgewell/proposal-logical-assignment, which I'll hopefully be [presenting in March](https://github.com/tc39/agendas/pull/318).

I'm bringing it [back](https://github.com/babel/babel/pull/516). 😉

### CODE REVIEW

File: packages/babylon/src/tokenizer/index.js [hunk 1]
```diff
@@ -464,9 +464,14 @@ export default class Tokenizer extends LocationParser {
     const next = this.input.charCodeAt(this.state.pos + 1);
 
     if (next === code) {
+      const assign =
+        this.hasPlugin("logicalAssignment") &&
+        this.input.charCodeAt(this.state.pos + 2) === charCodes.equalsTo;
```
Comment (nicolo-ribaudo) [inline packages/babylon/src/tokenizer/index.js]: Can you use `this.expectPlugin`?

Comment (jridgewell) [reply packages/babylon/src/tokenizer/index.js]: Done.

**Comment by hzoo** [comment]
I want to be able to test this in the REPL haha, should we add it to stage 0 or make a -1 lol..

**Comment by jridgewell** [comment]
I don't think stage 0 can be blocked, so my proposal is de-facto stage 0. I'll add it to the preset.

**Comment by hzoo** [comment]
Yeah -1 only relevant in that it hasn't been presented, but doesn't matter too much if it will be soon. Either way :+1:, I guess we should re-think how we signal stage x presets (given there's been a lot of talk about removing them entirely)

**Comment by nicolo-ribaudo** [comment]
Oh, I forgot to ask you to add this plug-in to https://github.com/babel/babel/blob/master/packages/babel-core/src/transformation/util/missing-plugin-helper.js

**Comment by jridgewell** [comment]
Done.

**Pull request merged**